### PR TITLE
[Merged by Bors] - Add checks to prevent fwding old messages

### PR DESCRIPTION
**Pull request by paulhauner**
## Issue Addressed

NA

## Proposed Changes

Checks to see if attestations or sync messages are still valid before "accepting" them for propagation.

## Additional Info

NA

### CODE REVIEW

**Review by AgeManning — APPROVED** [review]
Looks like a strict improvement to me.

File: beacon_node/network/src/beacon_processor/worker/gossip_methods.rs [hunk 1]
```diff
@@ -2096,4 +2100,50 @@ impl<T: BeaconChainTypes> Worker<T> {
...
+        &self,
+        sync_message_slot: Slot,
+        message_id: MessageId,
+        peer_id: PeerId,
+    ) {
+        let is_timely = self
+            .chain
+            .slot_clock
+            .now()
+            .map_or(false, |current_slot| sync_message_slot == current_slot);
```
Comment (michaelsproul) [inline beacon_node/network/src/beacon_processor/worker/gossip_methods.rs:2145]: Ahah, re-reading this made me realise I was wrong about the other PR: https://github.com/sigp/lighthouse/pull/2974#discussion_r796157356

**Review by michaelsproul — APPROVED** [review]
LGTM!

**Pull request closed without merging**